cookbook/profiler/data_collector.rst

```diff
@@ -46,106 +46,160 @@ access to in local properties.
-            <span>100 units</span>
-        </div>
-        <div class="sf-toolbar-info-piece">
-            <b>Another quick thing</b>
-            <span>300 units</span>
-        </div>
+            {# this is the content displayed when hovering the mouse over
+               the toolbar panel #}
+            <div class="sf-toolbar-info-piece">
+                <b>Quick piece of data</b>
+                <span>100 units</span>
+            </div>
+            <div class="sf-toolbar-info-piece">
+                <b>Another piece of data</b>
+                <span>300 units</span>
+            </div>
+        {% endset %}
+
+        {# the 'link' value set to 'false' means that this panel doesn't
+           show a section in the web profiler. #}
+        {{ include('@WebProfiler/Profiler/toolbar_item.html.twig', { link: false }) }}
+    {% endblock %}
+
+.. tip::
+
+    Built-in collector templates define all their images as embedded base64-encoded
+    images. This makes them work everywhere without having to mess with web assets
+    links:
+
+    .. code-block:: html
+
+        <img src="data:image/png;base64,..." />
+
+    Another solution is to define the images as SVG files. In addition to being
+    resolution-independent, these images can be easily embedded in the Twig
+    template or included from an external file to reuse them in several templates:
+
+    .. code-block:: jinja
+
+        {{ include('@App/data_collector/icon.svg') }}
+
+    You are encouraged to use the latter technique for your own toolbar panels.
+
+If the toolbar panel includes extended web profiler information, the Twig template
+must also define other blocks:
+
+.. code-block:: jinja
+
+    {% extends '@WebProfiler/Profiler/layout.html.twig' %}
+
+    {% block toolbar %}
+        {% set icon %}
+            <span class="icon"><img src="..." alt=""/></span>
+            <span class="sf-toolbar-status">Information</span>
         {% endset %}
 
-        {# Set the "link" value to false if you do not have a big "panel"
-           section that you want to direct the user to. #}
-        {{ include('@WebProfiler/Profiler/toolbar_item.html.twig', { 'link': true }) }}
+        {% set text %}
+            <div class="sf-toolbar-info-piece">
+                {# ... #}
+            </div>
+        {% endset %}
 
+        {# the 'link' value is now set to 'true', which allows the user to click
+           on it to access to the web profiler panel. Since 'true' is the default
+           value, you can omit the 'link' parameter entirely #}
+        {{ include('@WebProfiler/Profiler/toolbar_item.html.twig', { link: true }) }}
     {% endblock %}
 
     {% block head %}
-        {# Optional, if you need your own JS or CSS files. #}
-        {{ parent() }} {# Use parent() to keep the default styles #}
+        {# Optional, here you can link to or define your own CSS and JS contents #}
+        {# Use {{ parent() }} to keep the default styles #}
```

Comment:
I would leave this line as it was before, otherwise people won't get any styles if they copy/past the example